darts/datasets/__init__.py

```diff
@@ -511,3 +517,92 @@ def pre_proces_fn(extracted_dir, dataset_path):
                 pre_process_zipped_csv_fn=pre_proces_fn,
             )
         )
+
+
+class UberTLCDataset(DatasetLoaderCSV):
+    """
+    14.3 million Uber pickups from January to June 2015. The data is resampled to hourly or daily based sample_freq
+    on using the locationID as the target.
+    Source: [1]_
+
+    Loading this dataset will provide a multivariate timeseries with 262 columns for each locationID.
+    The following code can be used to convert the dataset to a list of univariate timeseries,
+    one for each locationID.
+
+    .. highlight:: python
+    .. code-block:: python
+
+        import pandas as pd
+        from darts import TimeSeries
+        from darts.datasets import UberTLCDataset
+
+        dataset_hourly = UberTLCDataset("hourly").load().pd_dataframe()
+
+        ts_list = []  # list of timeseries
+        for label in dataset_hourly:
+            srs = dataset_hourly[label]
+
+            # filter column down to the period of recording
+            start_date = min(srs.fillna(method="ffill").dropna().index)
+            end_date = max(srs.fillna(method="bfill").dropna().index)
+            active_range = (srs.index >= start_date) & (srs.index <= end_date)
+            srs = srs[active_range]
+
+            # convert to timeseries
+            tmp = pd.DataFrame({"locationID": srs})
+            tmp["date"] = tmp.index
+            ts = TimeSeries.from_dataframe(tmp, "date", ["locationID"])
+            ts_list.append(ts)
+
+    ..
+
+
+    References
+    ----------
+    .. [1] https://github.com/fivethirtyeight/uber-tlc-foil-response
+
+    """
+
+    def __init__(self, sample_freq="hourly"):
```

Comment:
How about a parameter specifying whether we want this dataset to return a single multivariate series, or multiple series (could be a multivariate one by default). That would be nice and also would remove the need to copy/paste some of the code from the docstring example in case someone needs the multiple univariate option.
We could even have such a parameter for all the multivariate series we currently have - I guess (hope? :D) that this would be simple and adding this parameter to the metadata parameters and adapting the DatasetLoaderCSV class with a code similar the one that you put in the docstring could be enough.

Reply:
Let me know what you think of the implementation. If it looks good, I will add it to the other series

Reply:
Yes, looks very good.